who_is_active.sql

```diff
@@ -2662,9 +2806,83 @@ BEGIN;
 					ELSE
 						'NULL '
 				END + 
-					'AS additional_info, 
-				x.start_time, 
+					'AS additional_info,
 					' +
+				CASE
+					WHEN
+						@output_column_list LIKE '%|[memory_grant_info|]%' ESCAPE '|'
+						AND @get_memory_grant_info = 1 THEN'
+						(
+							SELECT TOP(@i)
+							(
+								SELECT TOP(@i)	 
+									x.request_time,
+									x.grant_time,
+									x.wait_time_ms,
+									x.requested_memory AS requested_memory_kb,	
+									x.granted_memory AS granted_memory_kb,
+									x.used_memory AS used_memory_kb,	
+									x.max_used_memory AS max_used_memory_kb,
+									x.ideal_memory AS ideal_memory_kb,	
+									x.required_memory AS required_memory_kb,
+									x.queue_id AS queue_id,
+									x.wait_order AS wait_order,
+									x.is_next_candidate AS is_next_candidate,
+									x.dop,
+									CAST(x.query_subtree_cost AS NUMERIC(13,4)) AS query_subtree_cost
+								FOR XML 
+									PATH(''query_memory_grants''), 
+									TYPE
+							),
+							(
+								SELECT TOP(@i)
+									x.timeout_error_count,
+									x.target_memory_mb,
+									x.max_target_memory_kb,
+									x.total_memory_kb,
+									x.available_memory_kb,
+									x.granted_memory_kb,
+									x.used_memory_kb,
+									x.grantee_count,
+									x.waiter_count
+								FOR XML 
+									PATH(''query_resource_semaphores''), 
+									TYPE
+							),
+							(
+								SELECT TOP(@i)	
+									x.wg_name AS name,
+									x.request_max_memory_grant_percent,
+									x.request_max_cpu_time_sec,
+									x.request_memory_grant_timeout_sec,
+									x.max_dop
+								FOR XML 
+									PATH(''resource_governor_workload_groups''), 

```

Comment:
I think we should change this and the below resource pool section, to singular -- there will only ever be one of each for a given query, right? Also, I think we should drop "resource_governor_" from the names. Thoughts?

Likewise, how about if we drop "query_" from the names right above these?

Reply:
Right, a query would only have one resource pool ever.  Good catch 👍  Also I agree with the name changes.

Reply:
Okay. I'm doing some work in here so I'll handle this myself. If I can get my kids to leave me alone, maybe we can get this merged today. Otherwise, tomorrow. Almost there!

Reply:
Cool, looking forward to it!